src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -0,0 +1,1110 @@
+                type=>molecularProfileTypeDisplayOrder.indexOf(type)
+            ).map(type=>({
+                value: type,
+                label: molecularProfileTypeToDisplayType[type]
+            }));
+        } else {
+            return [];
+        }
+    }
+
+    @computed get profileNameOptionsByType() {
+        if (this.props.store.molecularProfilesInStudies.isComplete) {
+            return _.mapValues(
+                _.groupBy(this.props.store.molecularProfilesInStudies.result, profile=>profile.molecularAlterationType),
+                profiles=>profiles.map(p=>({value:p.molecularProfileId, label:p.name}))
+            );
+        } else {
+            return {};
+        }
+    }
+
+    private onVerticalAxisProfileTypeSelect(option:any) {
+        this.vertSelection.molecularProfileType = option.value;
+    }
+
+    private onHorizontalAxisProfileTypeSelect(option:any) {
+        this.horzSelection.molecularProfileType = option.value;
+    }
+
+    private onVerticalAxisProfileIdSelect(option:any) {
+        this.vertSelection.molecularProfileId = option.value;
+    }
+
+    private onHorizontalAxisProfileIdSelect(option:any) {
+        this.horzSelection.molecularProfileId = option.value;
+    }
+
+    private onVerticalAxisClinicalAttributeSelect(option:any) {
+        this.vertSelection.clinicalAttributeId = option.value;
+    }
+
+    private onHorizontalAxisClinicalAttributeSelect(option:any) {
+        this.horzSelection.clinicalAttributeId = option.value;
+    }
+
+    @action
+    private swapHorzVertSelections() {
+        for (const key of (Object.keys(this.horzSelection) as (keyof AxisMenuSelection)[])) {
+            const horzVersion = this.horzSelection[key];
+            this.horzSelection[key] = this.vertSelection[key];
+            this.vertSelection[key] = horzVersion;
+        }
+    }
+
+    @computed get bothAxesMolecularProfile() {
+        return (this.horzSelection.axisType === AxisType.molecularProfile) &&
+             (this.vertSelection.axisType === AxisType.molecularProfile);
+    }
+
+    @computed get sameGeneInBothAxes() {
+        return  this.bothAxesMolecularProfile &&
+            (this.horzSelection.entrezGeneId === this.vertSelection.entrezGeneId);
+    }
+
+    @computed get cnaPromise() {
+        if (this.viewType === ViewType.CopyNumber && this.props.store.studyToMolecularProfileDiscrete) {
+            if (!this.sameGeneInBothAxes) {
+                throw ("Plots tab invariant error: viewType is CopyNumber must mean that both axes are molecularProfile and have the same gene");
+            }
+            const entrezGeneId = this.horzSelection.entrezGeneId!;
+            const queries = _.values(this.props.store.studyToMolecularProfileDiscrete.result)
+                            .map(p=>({molecularProfileId: p.molecularProfileId, entrezGeneId}));
+            const promises = this.props.store.numericGeneMolecularDataCache.getAll(queries);
+            return remoteData({
```

Comment:
could this whole @computed by represented as a remoteData in the usual fashion?   when cnaPromise recomputes, I don't think that the remoteData of prior computations will every be garbage collected

Reply:
done